### ISSUE TITLE
Error uploading dataset with imaging tabs to ingest prod.

### ISSUE THREAD

**Issue by ami-day** [ebi-ait/dcp-ingest-central#462]
I am working on an imaging technology dataset, and I am getting an error importing the spreadsheet to ingest production. I think this is because of known problems with the imaging tabs, but it is difficult to interpret the error message I am getting.

submission in ingest production: https://contribute.data.humancellatlas.org/submissions/detail?uuid=3cf32c24-c2a8-4874-a413-aeb1ab66b2af

**Comment by gabsie** [comment]
Hey, what is necessary here to unblock? :) Asking for clarity..
Gabs

**Comment by Wkt8** [comment]
I think there is some dev work required to investigate errors why we currently cannot upload imaging tabs to prod

**Comment by gabsie** [comment]
Hey @jacobwindsor - maybe this one?

**Comment by gabsie** [comment]
@Wkt8  - Wei, can you please specify here (in @ami-day's absence) what is actually needed to accept the imaging metadata, because Jacob misses  information on what they need to check for?
Actually any info here is welcome, because it might be throwing errors for something unexpected..? Or we need validation for something and we don't have the validation rules for it

Gabs

**Comment by Wkt8** [comment]
I'm uncertain! I think the issue here is that uploading any sort of submission with imaging tabs causes errors for some reason we're not sure of. 

I don't think it's a validation issue for any of the ontologies but I think the issue here is 'something is breaking and we don't know what' because the submission is stuck and we can't view it.

**Comment by jacobwindsor** [comment]
It looks to me like someone has deleted the submission

**Comment by Wkt8** [comment]
There are two examples of spreadsheets with imaging tabs here - although they might not quite be complete for export! They also represent different techniques and hence different ways of using the imaging tabs. Feel free to reach out to Ami or myself if you have more questions! 

https://docs.google.com/spreadsheets/d/1e-vEqEQpdlOtQX5iXOQphq2DE-ky_bEf/edit#gid=1189631159 (Ami)
https://docs.google.com/spreadsheets/d/1w1QCcrKDzVc4liq73lI8WNUPPusqFe_Z/edit#gid=959036600 (Wei)

**Comment by jacobwindsor** [comment]
I have created two submissions for these files in staging:

- https://staging.contribute.data.humancellatlas.org/submissions/detail?uuid=de50f53f-eeb3-47e8-a3ab-ba504281948d
  - This one looks fine
- https://staging.contribute.data.humancellatlas.org/submissions/detail?uuid=4eb6b847-81b2-4a45-95a3-d940d54217f9
  - This one is not functioning but I think is due to an error in the spreadsheet
  - What is the difference between "imaging protocol - channel" and "imaging protocol - probe"? I can't see anything related to this in the schema. I think it is just this that is causing the problem because the other sheet is fine @Wkt8 @ami-day 


New tabs in sheets:

- Imaged specimen
- Imaging preparation protocol
- Imaging protocol
- Image file

**Comment by ofanobilbao** [comment]
@ami-day @Wkt8 @ESapenaVentura to have a separate conversation on this

**Comment by aaclan-ebi** [comment]
There were issues in the imaging schemas.

**Comment by ipediez** [comment]
@aaclan-ebi to work today on this

**Comment by aaclan-ebi** [comment]
![Screenshot 2021-10-29 at 14.13.33.png](https://images.zenhubusercontent.com/5b042c754b5806bc2bd426e2/f2659e5c-bbb9-433e-a1a1-0abe17a4f4bd)

There is a problem associating the rows of the _module worksheets_ (e.g. Imaging protocol - Probe) to the _concrete entity_ (Imaging protocol). If I remember correctly, the id column (imaging_protocol.protocol_core.protocol_id) of the concrete entity worksheet (Imaging Protocol) should be added in the module worksheet to specify which concrete entity it belongs to. I tried adding this column, however, I got another error:

![Screenshot 2021-10-29 at 14.18.00.png](https://images.zenhubusercontent.com/5b042c754b5806bc2bd426e2/ed493b11-1e6c-4367-8749-1e987e1941b1)

It looks like another check/validation was added before which conflicts with the implementation of importing module worksheets. This part needs to be fixed.

**Comment by amnonkhen** [comment]
@aaclan-ebi still working on the fix

**Comment by ESapenaVentura** [comment]
Conflict in implementation - 2 options, need to figure out what's next @aaclan-ebi

**Comment by gabsie** [comment]
This needs @aaclan-ebi 's suggestions proposed here, and a response fro the wranglers how to proceed.

**Comment by aaclan-ebi** [comment]
Meeting minutes from yesterday: https://docs.google.com/document/d/1Z4vqVfbyoPKpebbg2-LRmxcZd1yAA3cPWZJapZ8xQzs/edit

Decided to remove the validation check which is blocking the importing of modules. We'll do more investigation on the motivation/need for that validation added before.

**Comment by ofanobilbao** [comment]
@aaclan-ebi will finalise the changes and create the PR

**Comment by aaclan-ebi** [comment]
Hi @ami-day , @Wkt8 , the fix is already in staging.

@ami-day please note there are a lot of other issues (e.g. some files linked to missing cell suspension) in the `GSE171668_SingleCellAndSpatialAtlasCovid` spreadsheet that needs to be fixed but given a  [simplified version spreadsheet](https://app.zenhub.com/files/299258791/20a72573-c677-4b91-8e79-5bef6a48d8a4/download) (I removed some worksheets) I was able to import the imaging protocol modules. 

@Wkt8 @ESapenaVentura I didn't remove the validation I was saying I'd remove before. I did some more testing and it looks like it captures errors where you specified unknown columns in module worksheets. I was able to fix the issue without removing this validation.

![Screenshot 2021-11-10 at 12.35.36.png](https://images.zenhubusercontent.com/5b042c754b5806bc2bd426e2/67b1a7fd-44c5-418f-8d03-936f9c050bf8)

![Screenshot 2021-11-10 at 12.37.24.png](https://images.zenhubusercontent.com/5b042c754b5806bc2bd426e2/5f415402-a52c-4050-9628-bc764134afb4)


Who can test this in staging?

**Comment by aaclan-ebi** [comment]
I also added an error message when the concrete entity id is not specified in the module worksheet:

![Screenshot 2021-11-10 at 12.41.08.png](https://images.zenhubusercontent.com/5b042c754b5806bc2bd426e2/23bd4cb8-ea84-4dff-8688-91d73cd863b7)


If the highlighted column is missing, there will be an error message:

![Screenshot 2021-11-10 at 12.13.16.png](https://images.zenhubusercontent.com/5b042c754b5806bc2bd426e2/3a3f92ac-48dd-4646-80aa-ab9395f87bfd)

Please let me know if you have suggestions on the error message :) Thanks!

**Comment by ipediez** [comment]
@ESapenaVentura started testing this

**Comment by ofanobilbao** [comment]
@ESapenaVentura will test today as Elixir was not working yesterday

**Comment by ESapenaVentura** [comment]
I have been trying to test this but I get a linking error https://staging.contribute.data.humancellatlas.org/submissions/detail?uuid=add6b701-5e1e-45d7-9816-e09e9b357381&project=f5694942-f2b4-45b3-b277-e3b2b184c6d2

No entity seems to be linked, I have looked at the spreadsheet and it should be linked, which is pretty weird

I have used the simplified spreadsheet from above

**Comment by aaclan-ebi** [comment]
@ESapenaVentura tested the imaging spreadsheet in staging and encountered a lot of linking errors. He'lll try to retest again today.

**Comment by aaclan-ebi** [comment]
Yeah, there was an outage in core yesterday and that could have affected the linking/importing. Please retry when you get a chance, Thanks @ESapenaVentura !

**Comment by aaclan-ebi** [comment]
@Wkt8 and I chatted about the testing of this yesterday. A possible issue or missing feature we see is the linking of a probe module row to multiple imaging protocols (still need to be verified). I am uncertain if this was already supported before or an enhancement. A workaround could be just to duplicate the probe row and link it to the second imaging protocol that should use it.

**Comment by Wkt8** [comment]
I have tested the following: 
- linking imaging_protocol.probe to imaging_protocol (this works)
- linking imaging_protocol.channel to imaging_protocol (this works)

Linking a probe or channel to multiple imaging protocols via introducing separators "||" does not work. However, as stated above, duplicating the row and linking it to a second imaging protocol does work. 
This is fine as I do not predict we need to do this. 

Tested in staging. 
Test spreadsheet - based off of a modified version of Ami's spreadsheet - found here: https://drive.google.com/drive/folders/15MOmkLStw7AzSoK6sSEbdv_HJY0bZcZF
